components/ModEditFacilitySection.vue

```diff
@@ -304,18 +351,54 @@ onBeforeMount(async () => {
         await facilityStore.getFacilities()
     }
 
+    if (!healthcareProfessionalsStore.healthcareProfessionalsData) {
+        await healthcareProfessionalsStore.getHealthcareProfessionals()
+    }
+
     facilityStore.selectedFacilityId = id as string
 
     // Set the active screen and ensure the UI state is consistent
-    screenStore.setActiveScreen(ModerationScreen.EditFacility)
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditFacility)
 
     facilityStore.setSelectedFacilityData(facilityStore.selectedFacilityId)
 
     facilityStore.initializeFacilitySectionValues(facilityStore.selectedFacilityData)
 
+    await nextTick()
+    if (healthcareProfessionalsStore.healthcareProfessionalsData
```

Comment:
I think this can be merged with the map function to just be a one line assignment 
like `healthcareProfessionalsStore.healthcareProfessionalsData?.map(...)`

Reply:
ah, I see this method is reused. I would move this check ^ into the method.

Perhaps `sync` instead of `set` would be a better prefix for the method?